planning/behavior_path_planner/autoware_behavior_path_planner_common/test/test_traffic_light_utils.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ament_index_cpp/get_package_share_directory.hpp"
+#include "autoware/behavior_path_planner_common/data_manager.hpp"
+#include "autoware/behavior_path_planner_common/utils/traffic_light_utils.hpp"
+#include "autoware_test_utils/autoware_test_utils.hpp"
+#include "autoware_test_utils/mock_data_parser.hpp"
+
+#include <autoware/universe_utils/geometry/geometry.hpp>
+
+#include <autoware_perception_msgs/msg/detail/traffic_light_group__struct.hpp>
+#include <autoware_planning_msgs/msg/detail/lanelet_route__struct.hpp>
+#include <geometry_msgs/msg/detail/pose__struct.hpp>
+#include <geometry_msgs/msg/detail/twist__struct.hpp>
+#include <tier4_planning_msgs/msg/detail/path_with_lane_id__struct.hpp>
+
+#include <gtest/gtest.h>
+#include <lanelet2_core/Forward.h>
+#include <yaml-cpp/node/node.h>
+
+#include <limits>
+#include <memory>
+
+using autoware::behavior_path_planner::PlannerData;
+using autoware::behavior_path_planner::TrafficSignalStamped;
+using autoware_perception_msgs::msg::TrafficLightGroupArray;
+using autoware_planning_msgs::msg::LaneletRoute;
+using geometry_msgs::msg::Pose;
+
+const double epsilon = 1e-06;
+
+class TrafficLightTest : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    planner_data_ = std::make_shared<PlannerData>();
+    const auto test_data_file =
+      ament_index_cpp::get_package_share_directory("autoware_behavior_path_planner_common") +
+      "/test_data/test_traffic_light.yaml";
+    YAML::Node config = YAML::LoadFile(test_data_file);
+
+    set_current_pose(config);
+    set_route_handler(config);
+    set_traffic_signal(config);
+  }
+
+  void set_current_pose(YAML::Node config)
+  {
+    const auto self_odometry =
+      autoware::test_utils::parse<nav_msgs::msg::Odometry>(config["self_odometry"]);
+    planner_data_->self_odometry = std::make_shared<const nav_msgs::msg::Odometry>(self_odometry);
+  }
+
+  void set_route_handler(YAML::Node config)
+  {
+    const auto route = autoware::test_utils::parse<LaneletRoute>(config["route"]);
+    const auto intersection_map =
+      autoware::test_utils::make_map_bin_msg(autoware::test_utils::get_absolute_path_to_lanelet_map(
+        "autoware_test_utils", "intersection/lanelet2_map.osm"));
+    planner_data_->route_handler->setMap(intersection_map);
```

Comment:
you can use

```suggestion
    const auto map_path = autoware::test_utils::resolve_pkg_share_uri(config["map_path_uri"]);
    planner_data_->route_handler->setMap(intersection_map);
```

Reply:
Fixed in 8f7ffc04b153626634ed689fcbd791d36b4116fe